src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticReferenceConfidenceModel.java

```diff
@@ -0,0 +1,91 @@
+        this.samples = samples;
+        this.genotypingEngine = genotypingEngine;
+    }
+
+        /**
+         * Calculate the genotype likelihoods for the sample in pileup for being hom-ref contrasted with being ref vs. alt
+         *
+         * @param ploidy target sample ploidy.
+         * @param pileup the read backed pileup containing the data we want to evaluate
+         * @param refBase the reference base at this pileup position
+         * @param qual the min base quality for a read in the pileup at the pileup position to be included in the calculation
+         * @param hqSoftClips running average data structure (can be null) to collect information about the number of high quality soft clips
+         * @return a RefVsAnyResult genotype call.
+         */
+    @Override
+    public ReferenceConfidenceResult calcGenotypeLikelihoodsOfRefVsAny(final int ploidy,
+                                                                       final ReadPileup pileup,
+                                                                       final byte refBase,
+                                                                       final byte qual,
+                                                                       final MathUtils.RunningAverage hqSoftClips,
+                                                                       final boolean readsWereRealigned) {
+
+        final SomaticRefVsAnyResult result = new SomaticRefVsAnyResult();
+        final Map<String, List<GATKRead>> perSampleReadMap = new HashMap<>();
+        perSampleReadMap.put(samples.getSample(0), pileup.getReads());
+        final ReadLikelihoods<Allele> readLikelihoods = new ReadLikelihoods<>(samples, new IndexedAlleleList<>(Arrays.asList(Allele.create(refBase,true), Allele.NON_REF_ALLELE)), perSampleReadMap);
+        final Iterator<PileupElement> pileupIter = pileup.iterator();
+        for (int i = 0; i < pileup.size(); i++) {
+            final PileupElement element = pileupIter.next();
+            final boolean isAlt = readsWereRealigned ? isAltAfterAssembly(element, refBase) : isAltBeforeAssembly(element, refBase);
+            final double nonRefLikelihood;
+            if (isAlt) {
+                nonRefLikelihood = QualityUtils.qualToProbLog10(qual);
+                result.nonRefDepth++;
+            } else {
+                nonRefLikelihood = QualityUtils.qualToErrorProbLog10(qual) + MathUtils.LOG10_ONE_THIRD;
+                result.refDepth++;
+            }
+            readLikelihoods.sampleMatrix(0).set(0, i, nonRefLikelihood);
+        }
+        result.lods = genotypingEngine.somaticLog10Odds(readLikelihoods.sampleMatrix(0));
+        return result;
+    }
+
+    @Override
+    public void addGenotypeData(final ReferenceConfidenceResult result, final GenotypeBuilder gb) {
+        gb.attribute(GATKVCFConstants.TUMOR_LOD_KEY, ((SomaticRefVsAnyResult)result).lods.get(Allele.NON_REF_ALLELE));
```

Comment:
If I understand correctly, this means that the reference confidence "score" is the TLOD of the non-ref allele, which is a little strange because this score increases as the evidence for a non-ref allele increases.  Also, if the math is being done right then the integration test vcf values (and the default TLOD bands) of 1000 or more would imply extremely good evidence of variation in the reference confidence blocks.  I think some of my math comments above might help.

Reply:
I wanted to have the output be logically equivalent to the germline GQ, which is the Phred-scaled probability that the genotype call was wrong.  As someone who's looked at a lot of GVCFs, it's nice to skim down the GQ column and see that everything is going well because all the numbers are on the large side, meaning the calls are all confident whether each position is variant or not. Maybe I need a different INFO key for the ref blocks because the number I'm getting out of my ref conf model is the LOD between the call (reference) and anything that's not the call (non-ref).  REFLOD?